core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/dbcreator/DatabaseCreator.java

```diff
@@ -349,6 +350,30 @@ private void executeSQLScript() throws Exception {
         }
     }
 
+    private String getDatabaseTypeWithVersion(String databaseType) throws SQLException {
+        String databaseProductVersion = null;
+        try {
+            if (databaseType != null && databaseType.equals("mysql")) {
+                if (conn != null && (!conn.isClosed())) {
+                    DatabaseMetaData metaData = conn.getMetaData();
+                    databaseProductVersion = metaData.getDatabaseProductVersion();
+                    double versionWithMinorNumber = Double.parseDouble(databaseProductVersion.substring(0, 3));
+                    if (versionWithMinorNumber >= 5.7) {
+                        databaseType = databaseType + "-5.7+";
+                    }
+                }
+            }
+        } catch (NumberFormatException e) {
+            /*
+            If unable to get the database version to check whether the database type is mysql 5.7 or higher,
+            ignore the error and continue with the default script.
+             */
+            log.warn("Unable to parse the database product version: " + databaseProductVersion

```

Comment:
In a case where we cannot determine the version  exactly, I think it is better we throw an exception, rather than blindly executing the default mysql script, WDYT?